docs/framework/network-programming/how-to-list-directory-contents-with-ftp.md

```diff
@@ -1,55 +1,77 @@
+title: "How to: List directory contents with FTP"
+description: "This article shows a sample of how to list the directory contents of an FTP server."
+ms.date: "06/25/2018"
+dev_langs: 
+  - "csharp"
+  - "vb"
 ms.assetid: 130c64c9-7b7f-4672-9b3b-d946bd2616c5
-author: "mcleblanc"
-ms.author: "markl"
-manager: "markl"
 ---
-# How to: List Directory Contents with FTP
-This sample shows how to list the directory contents of an FTP server.  
-  
-## Example  
-  
-```csharp  
-using System;  
-using System.IO;  
-using System.Net;  
-using System.Text;  
-  
-namespace Examples.System.Net  
-{  
-    public class WebRequestGetExample  
-    {  
-        public static void Main ()  
-        {  
-            // Get the object used to communicate with the server.  
-            FtpWebRequest request = (FtpWebRequest)WebRequest.Create("ftp://www.contoso.com/");  
-            request.Method = WebRequestMethods.Ftp.ListDirectoryDetails;  
-  
-            // This example assumes the FTP site uses anonymous logon.  
-            request.Credentials = new NetworkCredential ("anonymous","janeDoe@contoso.com");  
-  
-            FtpWebResponse response = (FtpWebResponse)request.GetResponse();  
-  
-            Stream responseStream = response.GetResponseStream();  
-            StreamReader reader = new StreamReader(responseStream);  
-            Console.WriteLine(reader.ReadToEnd());  
-  
-            Console.WriteLine("Directory List Complete, status {0}", response.StatusDescription);  
-  
-            reader.Close();  
-            response.Close();  
-        }  
-    }  
-}  
-```  
-  
-## Compiling the Code  
- This example requires:  
-  
--   References to the **System.Net** namespace.  
-  
-## Robust Programming  
-  
-## .NET Framework Security
+# How to: List directory contents with FTP
+
+This sample shows how to list the directory contents of an FTP server.
+
+## Example
+
+```csharp
+using System;
+using System.IO;
+using System.Net;
+
+namespace Examples.System.Net
+{
+    public class WebRequestGetExample
+    {
+        public static void Main ()
+        {
+            // Get the object used to communicate with the server.
+            FtpWebRequest request = (FtpWebRequest)WebRequest.Create("ftp://www.contoso.com/");
+            request.Method = WebRequestMethods.Ftp.ListDirectoryDetails;
+
+            // This example assumes the FTP site uses anonymous logon.
+            request.Credentials = new NetworkCredential ("anonymous","janeDoe@contoso.com");
+
+            FtpWebResponse response = (FtpWebResponse)request.GetResponse();
+
+            Stream responseStream = response.GetResponseStream();
+            StreamReader reader = new StreamReader(responseStream);
+            Console.WriteLine(reader.ReadToEnd());
+
+            Console.WriteLine("Directory List Complete, status {0}", response.StatusDescription);
+
+            reader.Close();
+            response.Close();
+        }
+    }
+}
+```
+
+```vb
```

Comment:
Same comments as previously: System does not need to be imported, and a Module is better Irequiring a change to the Main method signature as well).